pubsub: TestConcurrentReceivesGetAllTheMessages fails intermittently

Found on Travis:

```
ok  	github.com/google/go-cloud/internal/batcher	3.904s	coverage: 1.0% of statements in ./...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xa6039f]
goroutine 1400 [running]:
github.com/google/go-cloud/internal/pubsub_test.TestConcurrentReceivesGetAllTheMessages.func1(0xc0003ac5a0, 0xc000317770, 0x1779180, 0xc00053e5c0, 0xc000309000, 0xc0003ac5b0, 0xc0003af770)
	C:/Users/travis/gopath/src/github.com/google/go-cloud/internal/pubsub/pubsub_test.go:157 +0x8f
created by github.com/google/go-cloud/internal/pubsub_test.TestConcurrentReceivesGetAllTheMessages
	C:/Users/travis/gopath/src/github.com/google/go-cloud/internal/pubsub/pubsub_test.go:146 +0x448
FAIL	github.com/google/go-cloud/internal/pubsub	2.199s
```

--- a/internal/pubsub/pubsub_test.go
+++ b/internal/pubsub/pubsub_test.go
@@ -152,6 +152,7 @@ func TestConcurrentReceivesGetAllTheMessages(t *testing.T) {
 						return
 					}
 					t.Error(err)
+					return
 				}
 				mu.Lock()
 				receivedMsgs[string(m.Body)]++

